components/script/dom/globalscope.rs

```diff
@@ -2848,6 +2849,28 @@ impl GlobalScope {
         is_js_evaluation_allowed == CheckResult::Allowed
     }
 
+    pub(crate) fn should_navigation_request_be_blocked(&self, url: &ServoUrl) -> bool {
+        let Some(csp_list) = self.get_csp_list() else {
+            return true;
```

Comment:
I just realised that this is wrong. It should return false instead. That's probably why all those tests fail.